local-cli/runAndroid/runAndroid.js

```diff
@@ -123,15 +123,28 @@ function buildAndRun(args, reject) {
 }
 
 function startServerInNewWindow() {
+  var yargV = require('yargs').argv;
+
   const launchPackagerScript = path.resolve(
     __dirname, '..', '..', 'packager', 'launchPackager.command'
   );
 
   if (process.platform === 'darwin') {
-    child_process.spawnSync('open', [launchPackagerScript]);
+    if (!yargV.open) {
+      return child_process.spawnSync('open', [launchPackagerScript]);
+    }
+    return child_process.spawnSync('open', ['-a', yargV.open, launchPackagerScript]);
```

Comment:
Sorry I just meant flipping the condition (and made a typo in the last comment :)):

if (yargV.open) {
   return ...
}
return ...   // The "default" case at the end